Build fails on FreeBSD 12.1-RELEASE

Following instructions in `doc/build-freebsd.md` encounters this error:

```
Build Options:
  with endomorphism       = no
  with ecmult precomp     = yes
  with external callbacks = no
  with jni                = no
  with benchmarks         = yes
  with coverage           = no
  module ecdh             = yes
  module recovery         = yes

  asm                     = x86_64
  bignum                  = no
  field                   = 64bit
  scalar                  = 64bit
  ecmult window size      = 15

  CC                      = gcc
  CFLAGS                  = -g -O2 -W -std=c89 -pedantic -Wall -Wextra -Wcast-align -Wnested-externs -Wshadow -Wstrict-prototypes -Wno-unused-function -Wno-long-long -Wno-overlength-strings -fvisibility=hidden -O3
  CPPFLAGS                = 
  LDFLAGS                 = 


Options used to compile and link:
  with wallet   = no
  with gui / qt = no
  with zmq      = yes
  with test     = yes
    with fuzz   = no
  with bench    = yes
  with upnp     = yes
  use asm       = yes
  liquid_build  = no
  sanitizers    = 
  debug enabled = no
  gprof enabled = no
  werror        = no

  target os     = 
  build os      = 

  CC            = gcc
  CFLAGS        = -g -O2
  CPPFLAGS      =   -U_FORTIFY_SOURCE -D_FORTIFY_SOURCE=2  -DHAVE_BUILD_INFO -D__STDC_FORMAT_MACROS
  CXX           = g++ -std=c++11
  CXXFLAGS      =   -fstack-reuse=none -Wstack-protector -fstack-protector-all  -Wall -Wextra -Wformat -Wvla -Wredundant-decls  -Wno-unused-parameter -Wno-implicit-fallthrough   -g -O2
  LDFLAGS       = -pthread  -Wl,-z,relro -Wl,-z,now -pie  
  ARFLAGS       = cr

Making all in src
gmake[1]: Entering directory '/home/wiz/elements/src'
gmake[2]: Entering directory '/home/wiz/elements/src'
  CXX      elementsd-bitcoind.o
  CXX      libbitcoin_server_a-addrdb.o
  CXX      libbitcoin_server_a-addrman.o
  CXX      libbitcoin_server_a-assetsdir.o
  CXX      libbitcoin_server_a-banman.o
  CXX      libbitcoin_server_a-bloom.o
  CXX      libbitcoin_server_a-blockencodings.o
  CXX      libbitcoin_server_a-blockfilter.o
  CXX      libbitcoin_server_a-block_proof.o
  CXX      libbitcoin_server_a-chain.o
  CXX      libbitcoin_server_a-checkpoints.o
  CXX      libbitcoin_server_a-confidential_validation.o
  CXX      libbitcoin_server_a-httprpc.o
  CXX      libbitcoin_server_a-httpserver.o
  CXX      libbitcoin_server_a-init.o
  CXX      libbitcoin_server_a-dbwrapper.o
  CXX      libbitcoin_server_a-mainchainrpc.o
httpserver.cpp:74:10: error: 'deque' in namespace 'std' does not name a template type
   74 |     std::deque<std::unique_ptr<WorkItem>> queue;
      |          ^~~~~
httpserver.cpp:33:1: note: 'std::deque' is defined in header '<deque>'; did you forget to '#include <deque>'?
   32 | #include <support/events.h>
  +++ |+#include <deque>
   33 | 
httpserver.cpp: In member function 'bool WorkQueue<WorkItem>::Enqueue(WorkItem*)':
httpserver.cpp:92:13: error: 'queue' was not declared in this scope; did you mean 'Enqueue'?
   92 |         if (queue.size() >= maxDepth) {
      |             ^~~~~
      |             Enqueue
httpserver.cpp:95:9: error: 'queue' was not declared in this scope; did you mean 'Enqueue'?
   95 |         queue.emplace_back(std::unique_ptr<WorkItem>(item));
      |         ^~~~~
      |         Enqueue
httpserver.cpp: In member function 'void WorkQueue<WorkItem>::Run()':
httpserver.cpp:106:35: error: 'queue' was not declared in this scope; did you mean 'Enqueue'?
  106 |                 while (running && queue.empty())
      |                                   ^~~~~
      |                                   Enqueue
httpserver.cpp:110:31: error: 'queue' was not declared in this scope; did you mean 'Enqueue'?
  110 |                 i = std::move(queue.front());
      |                               ^~~~~
      |                               Enqueue
  CXX      libbitcoin_server_a-merkleblock.o
```

I have same error on Arch linux.
```
git checkout elements-0.18.1.4
./autogen.sh
./configure --prefix=/home/kanna/elementsd/0.18.1.4 --with-gui=no --disable-wallet
make check
```
Then I get same error. `httpserver.cpp:74:10: error: 'deque' in namespace 'std' does not name a template type`

```
Build Options:
  with endomorphism       = no
  with ecmult precomp     = yes
  with external callbacks = no
  with jni                = no
  with benchmarks         = yes
  with coverage           = no
  module ecdh             = yes
  module recovery         = yes

  asm                     = x86_64
  bignum                  = no
  field                   = 64bit
  scalar                  = 64bit
  ecmult window size      = 15

  CC                      = gcc
  CFLAGS                  = -g -O2 -W -std=c89 -pedantic -Wall -Wextra -Wcast-align -Wnested-externs -Wshadow -Wstrict-prototypes -Wno-unused-function -Wno-long-long -Wno-overlength-strings -fvisibility=hidden -O3
  CPPFLAGS                = 
  LDFLAGS                 = 

Fixing libtool for -rpath problems.

Options used to compile and link:
  with wallet   = yes
  with gui / qt = no
  with zmq      = yes
  with test     = yes
    with fuzz   = no
  with bench    = yes
  with upnp     = no
  use asm       = yes
  liquid_build  = no
  sanitizers    = 
  debug enabled = no
  gprof enabled = no
  werror        = no

  target os     = linux
  build os      = 

  CC            = /usr/bin/ccache gcc
  CFLAGS        = -g -O2
  CPPFLAGS      =   -U_FORTIFY_SOURCE -D_FORTIFY_SOURCE=2  -DHAVE_BUILD_INFO -D__STDC_FORMAT_MACROS
  CXX           = /usr/bin/ccache g++ -std=c++11
  CXXFLAGS      =   -fstack-reuse=none -Wstack-protector -fstack-protector-all  -Wall -Wextra -Wformat -Wvla -Wredundant-decls  -Wno-unused-parameter -Wno-implicit-fallthrough   -g -O2
  LDFLAGS       = -pthread  -Wl,-z,relro -Wl,-z,now -pie  
  ARFLAGS       = cr
```

--- a/doc/build-freebsd.md
+++ b/doc/build-freebsd.md
@@ -41,6 +41,7 @@ export BDB_PREFIX="$PWD/db4"
 With wallet:
 ```shell
 ./autogen.sh
+MAKE=gmake CC=cc CXX=c++ CPPFLAGS=-I/usr/local/include \
 ./configure --with-gui=no \
     BDB_LIBS="-L${BDB_PREFIX}/lib -ldb_cxx-4.8" \
     BDB_CFLAGS="-I${BDB_PREFIX}/include"
@@ -49,6 +50,7 @@ With wallet:
 Without wallet:
 ```shell
 ./autogen.sh
+MAKE=gmake CC=cc CXX=c++ CPPFLAGS=-I/usr/local/include \
 ./configure --with-gui=no --disable-wallet
 ```
 

